### ISSUE TITLE
Not able to download v2.1.0 from maven repo

### ISSUE THREAD

**Issue by surendrajat** [DexPatcher/dexpatcher-gradle#32]
Was wondering if dexpatcher repo has been updated.

A side question, will the patches continue to work on if I want to use the latest v2.1.1+ given they're in maven repo as well? I see the release notes say apk library won't work.

**Comment by Lanchon** [comment]
dxp-gradle is published on the gradle plugin portal
https://plugins.gradle.org/search?term=dexpatcher

2.1.0 has been published there but it took some time to be approved
now it is

the rest of the stuff is in my own repo, and it is up to date

dxp-gradle 2.1.x versions will be on the portal once finalized

**Comment by surendrajat** [comment]
Not sure why but I had to change it to
```
dependencies {
    classpath "gradle.plugin.com.github.lanchon.dexpatcher:dexpatcher-gradle:2.1.0"
}
```
to make it work.

**Comment by Lanchon** [comment]
does it even work? it shouldn't! it seems u r not checking out the right tag

for dxp-gradle v2.1.0 you need to checkout samples v2.1.0

i recommend u use `git gui` and `gitk --all` if u r in doubt.

**Comment by surendrajat** [comment]
![Screenshot from 2020-06-24 11-53-58](https://user-images.githubusercontent.com/11215000/85508558-efaccd00-b611-11ea-96cd-5073a6ecd6a4.png)

Yes. I did this. Somehow it keeps throwing:
```  
A problem occurred configuring root project 'patched-app'.
> Could not resolve all artifacts for configuration ':classpath'.
   > Could not find com.github.lanchon.dexpatcher:dexpatcher-gradle:2.1.0.
     Searched in the following locations:
       - https://dl.google.com/dl/android/maven2/com/github/lanchon/dexpatcher/dexpatcher-gradle/2.1.0/dexpatcher-gradle-2.1.0.pom
       - https://dl.google.com/dl/android/maven2/com/github/lanchon/dexpatcher/dexpatcher-gradle/2.1.0/dexpatcher-gradle-2.1.0.jar
       - https://jcenter.bintray.com/com/github/lanchon/dexpatcher/dexpatcher-gradle/2.1.0/dexpatcher-gradle-2.1.0.pom
       - https://jcenter.bintray.com/com/github/lanchon/dexpatcher/dexpatcher-gradle/2.1.0/dexpatcher-gradle-2.1.0.jar
       - https://plugins.gradle.org/m2/com/github/lanchon/dexpatcher/dexpatcher-gradle/2.1.0/dexpatcher-gradle-2.1.0.pom
     Required by:
         project :

```
but works if I use `classpath "gradle.plugin.com.github.lanchon.dexpatcher:dexpatcher-gradle:2.1.0"`. Weird

**Comment by Lanchon** [comment]
oh sorry! i misunderstood! yeah the gradle people prepended `gradle.plugin.` i'll remove the build and talk to them.

**Comment by Lanchon** [comment]
FYI...

https://github.com/iBotPeaches/Apktool/issues/2369

**Comment by surendrajat** [comment]
Yeah. Saw that in the morning as I'm following that project too :)
Cool and quick response. Hopefully, the next version is usable with the latest plugin for now at least.

**Comment by Lanchon** [comment]
so the gradle plugin portal publishing plugin, the one you MUST run locally on your PC in order to publish a plugin to the gradle plugin portal, is sneakily closed source and proprietary. i say sneakily because gradle presents themselves to be an open source solution and their docs link to the plugin. there is (or that may have been fixed in the last couple of days) absolutely no mention of the closed source nature of the publish plugin or its license anywhere. in order to realize it is closed-source you must go through all published sources and rule out it not being there -quite a job- and then ask gradle employees, and then maybe they'll tell you it is not free software.

so i was tricked into running proprietary software on my machines. horrible.

of course i am not using the portal -which misbehaved and mispublished my stuff anyway- again, so the newer plugins will need to find another home. also i reported and complained about this situation, so i got banned from the gradle org in github, lol.

so now i'm trying to get gradle to delete my stuff from their servers, but they are not replying to my emails, although i'm writing to the right address for that. and i can't open support tickets because, guess, yep they are hosted on github, they same place where they so hastily ban dissence.

that's why you haven't seen any progress these last weeks. needless to say, i'll avoid the gradle team and its products as much as possible in the future.

**Comment by surendrajat** [comment]
That's very unfortunate and deceiving. However, this is a trend in all profit-making companies to brand their semi-OpenSource software as OpenSource these days.

But have you explored the [GitHub packages](https://docs.github.com/en/packages/using-github-packages-with-your-projects-ecosystem/configuring-gradle-for-use-with-github-packages) option for just this?

**Comment by Lanchon** [comment]
> But have you explored the GitHub packages option for just this?

thanks!

interesting... i think i might extend my own repo with the plugins. the downside is that source will not be in the repo. the upside is that i can delete packages as i see fit. this helps especially with the first repo publications, which might have problems.

**Comment by 1LineAtaTime** [comment]
yeah, I had to change the following things to get the patched-app to compile:

`    dependencies {
        // Android Gradle plugins:
        classpath 'com.android.tools.build:gradle:3.4.3'
    }`

`
    dependencies {
        // DexPatcher Gradle plugins:
        classpath 'gradle.plugin.io.github.dexpatcher.dexpatcher-gradle:dexpatcher-gradle:2.1.0'
    }`

`apply plugin: 'io.github.dexpatcher.patched-application'`

**Comment by Lanchon** [comment]
ive requested gradle more than 2 weeks ago that they delete that plugin. please dont use it. it will stop working and the presence of usage statistics make it more difficult have gradle finally delete the thing. if u must use this unofficial 2.1.0, then download it and use it locally.

**Comment by 1LineAtaTime** [comment]
Got it. Thanks.
I believe I had downloaded the latest version with 2.1.3 (latest commit on Jun 23) --> 'com.github.lanchon.dexpatcher:dexpatcher-gradle:2.1.3'
It was giving me errors with not finding different repositories, so I had to change it.

**Comment by Lanchon** [comment]
checkout the V2.0.0 tag from git or download the tag's zip offered by github.

**Comment by surendrajat** [comment]
@Lanchon Just fyi... apktool v2.5.0 got released :)

**Comment by Lanchon** [comment]
i added it to my repo a day ago.

**Comment by surendrajat** [comment]
Hi @Lanchon.
I was wondering if you could release the latest version somewhere. I'm having some issues(like [this](https://stackoverflow.com/questions/65182767/i-the-error-unexpected-element-queries-found-in-manifest-how-to-fix-it) during manifest merge) with the old gradle plugin.
